internal/provider/baremetal/installConfig.go

```diff
@@ -40,7 +41,18 @@ func (p baremetalProvider) AddPlatformToInstallConfig(
 			p.Log.Warnf("Failed to unmarshall Host %s inventory", hostutil.GetHostnameForMsg(host))
 			return err
 		}
-		hosts[yamlHostIdx].BootMACAddress = inventory.Interfaces[0].MacAddress
+
+		for _, iface := range inventory.Interfaces {
+			if iface.MacAddress != "" {
```

Comment:
Maybe you could check the type? empty string or physical to be on the safe side

Reply:
I suppose if you look for empty string it would work, but the problem we're trying to solve with this PR is that we can't get the agent PR that sends the type data to pass e2e tests without a fix here first.

Plus I don't think we'd want to filter here. We should eventually be able to use whatever interface the user chooses, even if it's virtual, right? That said, this wasn't driven by a user selection before, which is why I asked why it was okay to just use the first in the list.

Reply:
I think we'd want to revisit this code when adding the environment variable to enable virtual interfaces ([MGMT-10411](https://issues.redhat.com/browse/MGMT-10411))

Reply:
i want to keep the same behavior as we have now, the empty string is for the type and it's targeting old agents. i think that the code looks good now.